modules/apps/notification/notification-web/src/main/resources/META-INF/resources/js/components/SettingsContainer/rolesUtil.ts

```diff
@@ -76,6 +76,20 @@ export async function getRoles() {
 	return ((await response.json()) as Roles).items;
 }
 
+export function uncheckMultiSelectItemChildrens(items: MultiSelectItem[]) {
+	return items.map((item) => {
+		return {
+			...item,
+			children: item.children.map((child) => {
+				return {
+					...child,
+					checked: false,
+				};
+			}),
+		};
+	});
+}
+
```

Comment:
Hey, @vngoliveira good job! What do you think about applying this pattern of unchecking role items also in the Primary Recipients component, with the aim of keeping both components working in the same way.

Cleaning the roles only in secondary recipients can give the impression of a bug if the same behavior is not presented in primary recipients.